### ISSUE TITLE
Does there any plan to support Chinese voice?

### ISSUE THREAD

**Issue by jinfagang** [r9y9/deepvoice3_pytorch#123]
Does there any plan to support Chinese voice ?

**Comment by jarred1989** [comment]
it support Chinese, but the wav synthsized by deepvoice3 is not as good as Tacotron

**Comment by jinfagang** [comment]
@jarred1989 Do u have any source that support Chinese synthetic with Tacotron?

**Comment by Martin-Laclaustra** [comment]
@jinfagang There is a repo that trained tacotron and DCTTS (the equivalent to nyanko here) for Chinese:
https://github.com/Kyubyong/css10
It would be nice to further develop this repo to make it more multilingual. I am preparing some pull requests in that direction.
@jarred1989 Could you share the code you wrote to test this repo in Chinese and your results?

**Comment by jarred1989** [comment]
@jinfagang You may have a look at this repo https://github.com/Rayhane-mamah/Tacotron-2
@Martin-Laclaustra Sorry  I cannot share code or results, but the mainly difference between Chinese and English maybe input symbols of the network, and we usually transform Chinese characters into Chinese pinyin while preprocessing the data. And you may refer to this issue https://github.com/keithito/tacotron/issues/118

**Comment by ywwhxxtw** [comment]
@jinfagang Hi, you may check [my repo](https://github.com/ywwhxxtw/deepvoice3_pytorch) for Chinese deepvoice3. It is still on optimizing. For now I tried deepvoice3 builder and I get a similar result as the JP and KO models (still with robotic sound).

**Comment by jinfagang** [comment]
@ywwhxxtw  Sounds very nice !! Did you got any synthsis samples for Chinese? Really like to see the result

**Comment by ywwhxxtw** [comment]
I guess, I may upload some good examples in a few days. But I'm afraid I can not offer the dataset since its a private one.

**Comment by jinfagang** [comment]
@ywwhxxtw  that's ok, There're some good enough opensource Chinese voice data though. But better normalize the data process pipeline for better training on the model

**Comment by ywwhxxtw** [comment]
@jinfagang Hi there, I'm glad to tell that I've got some pretty good samples, already uploaded them. Feel free to check it out.

**Comment by jinfagang** [comment]
@ywwhxxtw  That was excited!!

**Comment by jinfagang** [comment]
Here is I previous trained on tacotron result of Chinese:
[step-30000-audio.zip](https://github.com/r9y9/deepvoice3_pytorch/files/3185651/step-30000-audio.zip)

**Comment by lin88lin8850** [comment]
> @jinfagang Hi there, I'm glad to tell that I've got some pretty good samples, already uploaded them. Feel free to check it out.

Hi, how many days have u trained to get those good samples? Thank you in advance!

**Comment by jinfagang** [comment]
@lin88lin8850  this implementation seems not work on inference step. the generates one is over fit

**Comment by ywwhxxtw** [comment]
@lin88lin8850 I guess around 3 days and I get aligned model around 1 day. However after those good samples I can not move any further which means there is still robotic sounds. If anyone have any idea about this, you may consider share your idea.

**Comment by lin88lin8850** [comment]
> @lin88lin8850 this implementation seems not work on inference step. the generates one is over fit



> @lin88lin8850 I guess around 3 days and I get aligned model around 1 day. However after those good samples I can not move any further which means there is still robotic sounds. If anyone have any idea about this, you may consider share your idea.

Have u considered about replacing vocoder, like wavenet, waveRNN .etc?

**Comment by lin88lin8850** [comment]
> @ywwhxxtw Have u considered about replacing vocoder, like wavenet, waveRNN .etc?

**Comment by lin88lin8850** [comment]
@jinfagang yep, the alignment on test audio is bad, I can not generate any voice 
[
![step000300000_text2_multispeaker1_alignment](https://user-images.githubusercontent.com/26682403/60067737-692f8800-973e-11e9-9901-76637b1b1abe.png)
](url)

**Comment by hutuxiansheng** [comment]
> @jinfagang Hi there, I'm glad to tell that I've got some pretty good samples, already uploaded them. Feel free to check it out.

How do I train Chinese models myself，Can you provide relevant steps?

**Comment by YihuaLiang95** [comment]
> @lin88lin8850 I guess around 3 days and I get aligned model around 1 day. However after those good samples I can not move any further which means there is still robotic sounds. If anyone have any idea about this, you may consider share your idea.

Hi, I fail to get aligned model after 6 days training. I followed settings in your repo but results after 500k steps are just noise.... Could you please give me some advice?

**Comment by ccuiyuhan** [comment]
> @jinfagang Hi, you may check [my repo](https://github.com/ywwhxxtw/deepvoice3_pytorch) for Chinese deepvoice3. It is still on optimizing. For now I tried deepvoice3 builder and I get a similar result as the JP and KO models (still with robotic sound).

Thanks for your repo，and I ran your code，but，i met some problems，when i ran train.py, the loss is nan always. i dont konw what is the matter. how to slove it ? Hope some advice! Thanks!

**Comment by ccuiyuhan** [comment]
> @jinfagang Hi, you may check [my repo](https://github.com/ywwhxxtw/deepvoice3_pytorch) for Chinese deepvoice3. It is still on optimizing. For now I tried deepvoice3 builder and I get a similar result as the JP and KO models (still with robotic sound).

I debug the code found the attetion_loss=0, I found the attention module , there are code following:
        mask_value = -float("inf")
        if mask is not None:
            mask = mask.view(query.size(0), 1, -1)
            x.data.masked_fill_(mask, mask_value)
and i found the x became nan after the code, i dont kjnow how to slove it! Hope the answer!

**Comment by JohnHerry** [comment]
@jinfagang  Hi, I am try to train with Chinese dataset,  I am using multi-speaker dataset of MAGICDATA with 1016 speakers.  Total training steps upto 6000K, but the test tts result do not sound good. while the training process samples are good.  What is the problem?  
This problem is pointed out in  issure 201
